pkg/kubelet/kubelet.go

```diff
@@ -2071,6 +2088,14 @@ func (kl *Kubelet) SyncTerminatingPod(_ context.Context, pod *v1.Pod, podStatus
 		return fmt.Errorf("detected running containers after a successful KillPod, CRI violation: %v", runningContainers)
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.RestartContainerDuringTermination) {
+		// Once the containers are stopped, we can stop probing for liveness and readiness.
+		// TODO: once a pod is terminal, certain probes (liveness exec) could be stopped immediately after
+		//   the detection of a container shutdown or (for readiness) after the first failure. Tracked as
+		//   https://github.com/kubernetes/kubernetes/issues/107894 although may not be worth optimizing.
+		kl.probeManager.RemovePod(pod)
```

Comment:
@gjkim42 I think it's enough to move it here to keep probing during shutdown

Reply:
I think this was already being executed after the killPod (or syncTerminatingPod). Are there other calls that you want to be executed before this?

Reply:
yes I think the tests just above are important to make that work... I'm going to test that